packages/client/src/modals/TransactionSuccessModal.js

```diff
@@ -0,0 +1,134 @@
+import React, { useContext, useEffect, useState } from "react";
+import { Web3Context } from "contexts/Web3";
+import { useClipboard, useContacts } from "../hooks";
+import { Copy, ArrowUp, OpenNewTab } from "../components/Svg";
+import { getFlowscanUrlForTransaction, formatAddress } from "utils";
+import { ACTION_TYPES } from "../constants/enums";
+import { COIN_TYPE_TO_META, CONTRACT_NAME_TO_COIN_TYPE } from "../constants/maps";
+
+const TransactionSuccessModal = ({ actionData, txID, onClose, safeName, safeAddress }) => {
+    const { getNFTReference } = useContext(Web3Context);
+    const clipboard = useClipboard();
+    const [image, setImage] = useState();
+    const { contacts } = useContacts(safeAddress);
+
+    const { recipient, nftId, collectionId, vaultId, tokenAmount } = actionData;
+    const { displayName, icon } = getTokenMeta(vaultId) || {};
+    const { name: imageName, imageURI } = image || {};
+
+    const actionType = actionData.type;
+    
+    const NFTName = collectionId?.split(".")[2];
+    const NFTAddress = collectionId?.split(".")[1];
+
+    useEffect(() => {
+        if (actionType === ACTION_TYPES.TRANSFER_NFT) {
+            const getImageURL = async () => {
+                const result = await getNFTReference(NFTName, formatAddress(NFTAddress), recipient, nftId);
+                setImage(result);
+            };
+            getImageURL().catch(console.error);
+        }
+    }, [actionType, NFTName, NFTAddress, nftId, recipient, getNFTReference])
+
+    function getTokenMeta(vaultId) {
```

Comment:
suggest moving this to util, also create a similar function for collectionIds

Reply:
Moved